osquery/logger/plugins/aws_kinesis.cpp

```diff
@@ -65,41 +65,69 @@ Status KinesisLoggerPlugin::logString(const std::string& s) {
 
 Status KinesisLogForwarder::send(std::vector<std::string>& log_data,
                                  const std::string& log_type) {
-  std::vector<Aws::Kinesis::Model::PutRecordsRequestEntry> entries;
-  if (FLAGS_aws_kinesis_random_partition_key) {
-    boost::uuids::uuid uuid = boost::uuids::random_generator()();
-    partition_key_ = boost::uuids::to_string(uuid);
-  }
-  for (const std::string& log : log_data) {
-    if (log.size() > kKinesisMaxLogBytes) {
-      LOG(ERROR) << "Kinesis log too big, discarding!";
+  int retry_count = 100;
+  int retry_delay = 3000;
+  std::vector<std::string> batch = log_data;
+  // exit if we have sent all the data
+  while (batch.size() > 0) {
+    if (FLAGS_aws_kinesis_random_partition_key) {
+      boost::uuids::uuid uuid = boost::uuids::random_generator()();
+      partition_key_ = boost::uuids::to_string(uuid);
+    }
+    std::vector<Aws::Kinesis::Model::PutRecordsRequestEntry> entries;
+    for (const std::string& log : batch) {
+      if (log.size() > kKinesisMaxLogBytes) {
+        LOG(ERROR) << "Kinesis log too big, discarding!";
+      }
+      Aws::Kinesis::Model::PutRecordsRequestEntry entry;
+      entry.WithPartitionKey(partition_key_)
+          .WithData(Aws::Utils::ByteBuffer((unsigned char*)log.c_str(),
+                                           log.length()));
+      entries.push_back(std::move(entry));
     }
-    Aws::Kinesis::Model::PutRecordsRequestEntry entry;
-    entry.WithPartitionKey(partition_key_)
-        .WithData(
-            Aws::Utils::ByteBuffer((unsigned char*)log.c_str(), log.length()));
-    entries.push_back(std::move(entry));
-  }
-
-  Aws::Kinesis::Model::PutRecordsRequest request;
-  request.WithStreamName(FLAGS_aws_kinesis_stream)
-      .WithRecords(std::move(entries));
 
-  Aws::Kinesis::Model::PutRecordsOutcome outcome = client_->PutRecords(request);
-  Aws::Kinesis::Model::PutRecordsResult result = outcome.GetResult();
-  if (result.GetFailedRecordCount() != 0) {
-    for (const auto& record : result.GetRecords()) {
-      if (!record.GetErrorMessage().empty()) {
+    Aws::Kinesis::Model::PutRecordsRequest request;
+    request.WithStreamName(FLAGS_aws_kinesis_stream)
+        .WithRecords(std::move(entries));
+
+    Aws::Kinesis::Model::PutRecordsOutcome outcome =
+        client_->PutRecords(request);
+    Aws::Kinesis::Model::PutRecordsResult result = outcome.GetResult();
+    VLOG(1) << "Successfully sent "
+            << result.GetRecords().size() - result.GetFailedRecordCount()
+            << " of " << result.GetRecords().size() << " logs to Kinesis.";
+    if (result.GetFailedRecordCount() != 0) {
+      std::vector<std::string> resend;
+      std::string error_msg = "";
+      int i = 0;
```

Comment:
Checking the api docs on this, but I don't think so.  This was just carry over code from the previous functionality.